tests/edwardstic.py

```diff
@@ -0,0 +1,315 @@
+        # GIVEN
+        self.ca.set_pv_value("TURBO:START", "On", wait=True)
+
+        # THEN
+        self.ca.assert_that_pv_is("TURBO:STA", "Running")
+
+    def test_GIVEN_standby_mode_switched_on_WHEN_status_requested_THEN_standby_reads_switched_on(self):
+        # GIVEN
+        self.ca.set_pv_value("TURBO:SETSTBY", "Yes", wait=True)
+
+        # THEN
+        self.ca.assert_that_pv_is("TURBO:STBY", "Yes")
+
+    def test_GIVEN_standby_mode_switched_off_WHEN_status_requested_THEN_standby_reads_switched_off(self):
+        # GIVEN
+        self.ca.set_pv_value("TURBO:SETSTBY", "No", wait=True)
+
+        # THEN
+        self.ca.assert_that_pv_is("TURBO:STBY", "No")
+
+    @parameterized.expand([
+        ("turbo_status", "TURBO:STA"),
+        ("turbo_speed", "TURBO:SPEED"),
+        ("turbo_power", "TURBO:POWER"),
+        ("turbo_norm", "TURBO:NORM"),
+        ("turbo_standby", "TURBO:STBY"),
+        ("turbo_cycle", "TURBO:CYCLE")
+    ])
+    def test_GIVEN_disconnected_device_THEN_record_redirected_PVs_read_invalid(self, _, base_pv):
+        # GIVEN
+        self._lewis.backdoor_set_on_device("is_connected", False)
+
+        # WHEN
+        self.ca.assert_that_pv_alarm_is(base_pv, self.ca.Alarms.INVALID, timeout=20)
+        self.ca.assert_that_pv_alarm_is("{base}:ALERT".format(base=base_pv), self.ca.Alarms.INVALID)
+        self.ca.assert_that_pv_alarm_is("{base}:PRI".format(base=base_pv), self.ca.Alarms.INVALID)
+
+
+class TurboStatusTests(EdwardsTICBase, unittest.TestCase):
+    def get_base_PV(self):
+        return "TURBO:STA"
+
+    def get_alert_PV(self):
+        return "TURBO:STA:ALERT"
+
+    def get_priority_PV(self):
+        return "TURBO:STA:PRI"
+
+    def get_status_setter(self):
+        return "set_turbo_pump_state"
+
+    def get_alert_function(self):
+        return "set_turbo_alert"
+
+    def get_priority_function(self):
+        return "set_turbo_priority"
+
+    def get_status_labels():
+        return [("stopped", "Stopped", ChannelAccess.Alarms.NONE),
+                ("starting_delay", "Starting Delay", ChannelAccess.Alarms.NONE),
+                ("accelerating", "Accelerating", ChannelAccess.Alarms.NONE),
+                ("running", "Running", ChannelAccess.Alarms.NONE),
+                ("stopping_short_delay", "Stopping Short Delay", ChannelAccess.Alarms.NONE),
+                ("stopping_normal_delay", "Stopping Normal Delay", ChannelAccess.Alarms.NONE),
+                ("fault_braking", "Fault Breaking", ChannelAccess.Alarms.MAJOR),
+                ("braking", "Braking", ChannelAccess.Alarms.NONE)]
+
+
+class Gauge1Tests(GaugeTestBase, unittest.TestCase):
+    def get_gauge_number(self):
+        return 1
+
+
+class Gauge2Tests(GaugeTestBase, unittest.TestCase):
+    def get_gauge_number(self):
+        return 2
+
+
+class Gauge3Tests(GaugeTestBase, unittest.TestCase):
+    def get_gauge_number(self):
+        return 3
```

Comment:
Usually we would just do this with `@parameterized.expand`, maybe in combination with `itertools.product` if we also want to parameterize some other parameter at the same time. This is fine here though and does not need changing - just a note for future.